src/plugins/intel_cpu/src/emitters/snippets/x64/kernel_executors/brgemm_base.cpp

```diff
@@ -0,0 +1,274 @@
+// Copyright (C) 2020-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "brgemm_base.hpp"
+
+#include "common/utils.hpp"
+#include "dnnl_extension_utils.h"
+#include "transformations/snippets/x64/op/brgemm_cpu.hpp"
+#include "transformations/snippets/x64/op/brgemm_utils.hpp"
+
+#define DIM_CAST(X) static_cast<dnnl_dim_t>(X)
+#define DTYPE_CAST(X) static_cast<dnnl_data_type_t>(DnnlExtensionUtils::ElementTypeToDataType(X))
+
+using namespace Xbyak;
+using namespace dnnl::impl;
+using namespace dnnl::impl::cpu::x64;
+
+namespace ov {
+namespace intel_cpu {
+
+bool BrgemmBaseKernelConfig::is_completed() const {
+    return !utils::one_of(0, m_M, m_N, m_K, m_LDA, m_LDB, m_LDC) || is_empty();
+}
+
+bool BrgemmBaseKernelConfig::is_empty() const {
+    return everyone_is(0, m_M, m_N, m_K, m_LDA, m_LDB, m_LDC, m_beta);
+}
+
+bool BrgemmBaseKernelConfig::operator==(const BrgemmBaseKernelConfig& rhs) const {
+#define EQ(X) X == rhs.X
+    return EQ(m_hash) && EQ(m_beta) &&
+           EQ(m_M) && EQ(m_N) && EQ(m_K) &&
+           EQ(m_LDA) && EQ(m_LDB) && EQ(m_LDC) &&
+           (EQ(get_static_params().get()) || *get_static_params() == *(rhs.get_static_params()));
```

Comment:
Minor: we don't really need need `get` here.
Also: `EQ` and `HASH` are used in several blocks across this cpp. Should we define them once for the file instead of this `#define + #undef` blocks. This would make it easier to read